inc/saiport.h

```diff
@@ -74,6 +74,57 @@ typedef enum _sai_port_oper_status_t
 
 } sai_port_oper_status_t;
 
+/**
+ * @brief Attribute data for #SAI_PORT_ATTR_ERROR_STATUS
+ * Note enum values must be powers of 2 to be used as Bit mask to query multiple errors
+ *
+ * @flags free
+ */
+typedef enum _sai_port_error_status_t
+{
+    /** No errors */
+    SAI_PORT_ERROR_STATUS_CLEAR = 0,
```

Comment:
@prgeor @lguohan  I added exception for that, but i would vote that this flag should be removed, since its not real flag, usually flags are used like this:
```C
 if (x & FLAG) == FLAG)
```
checks if flag is set, in this case  if you use CLEAR value then this condidtion will be always true, and CLEAR is not part of actual flags that can be set, it seems like intention is to do this:
```C
if (x == CLEAR) ...
```
but i think that is a bad design, more natural is first code snipper or eventually it could be done like this:
```C
if (x) ... // some flags are set
```

do we need a live discussion on this or i can just remove this entry ?

Reply:
@kcudnik `sai_port_error_status_t` being used as enum or flag is up to the implementation. I can use enum value like below:-

I can use the enum value like so

`if (port_error_status)` {
    // Found Errors
}
else {
   // No Errors 
}
`

In another implementation I can look for only interested errors where flag could be useful.

Reply:
it's the same as i wrote in my comment, but there eis no need for SAI_PORT_ERROR_STATUS_CLEAR enum value zero, im proposing to remove it

Reply:
@kcudnik why do you say there is no need for `SAI_PORT_ERROR_STATUS_CLEAR ` ? I am planning to use it when there is port UP event and gave the example above .

Reply:
because that enum is not an actual flag

Reply:
@kcudnik as I mentioned I am planning to use `sai_port_error_status_t` as both as an enum value and also also as a flag. I don't want to check each and every flag rather at time I am interested only in few flags in which case I know the enum value and in some case I may be interested only to check one flag. It not just about CLEAR flag.

Reply:
CLEAR is not a flag is definition of absence of any flags,, read my first comment in this thread